### PR TITLE
Fix AWS Security Group leakage at creation when an error occures

### PR DESCRIPTION
**Pull request by vflaux**
fix aws-controllers-k8s/community#1990

Description of changes: 
Change the hook `security_group/sdk_create_post_set_output.go.tpl` to always return the resource even if an error occurred. At this point the Security Group exists and we should keep track of it.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ack_generate_info:
-  build_date: "2024-01-15T16:33:37Z"
-  build_hash: 3753ca3f6610172e9e652d6e5e62e0a05f2e639c
-  go_version: go1.21.5
-  version: v0.28.0-11-g3753ca3
+  build_date: "2024-01-16T14:28:05Z"
+  build_hash: 610c476ce898de19ef2552612761a0ed699fc444
+  go_version: go1.21.4
+  version: v0.28.0-14-g610c476-dirty
```
Comment (a-hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: Did you make any changes to the code-generator? If needed, we'll have to merge those as well.

File: pkg/resource/dhcp_options/sdk.go [hunk 2]
```diff
@@ -321,6 +321,7 @@ func (rm *resourceManager) sdkDelete(
 	defer func() {
 		exit(err)
 	}()
+
```
Comment (a-hilaly) [inline pkg/resource/dhcp_options/sdk.go:324]: I'm wondering how did we get extra new line.

File: templates/hooks/security_group/sdk_create_post_set_output.go.tpl [hunk 3]
```diff
@@ -5,18 +5,18 @@
...
-		return nil, err
+		return &resource{ko}, err
 	}
 
 	// A ReadOne call for SecurityGroup Rules (NOT SecurityGroups)
 	// is made to refresh Status.Rules with the recently-updated
 	// data from the above `sync` call
 	if rules, err := rm.getRules(ctx, &resource{ko}); err != nil {
-		return nil, err
+		return &resource{ko}, err
```
Comment (a-hilaly) [inline templates/hooks/security_group/sdk_create_post_set_output.go.tpl:19]: 👍

**Review by a-hilaly — COMMENTED** [review]
Great catch @vflaux !
/ok-to-test
/test all

**Comment by vflaux** [comment]
I used the main branch of code-generator without modifications to regen the code.
Should I use the v0.28.0-11-g3753ca3 tag instead so I do not introduce unrelated changes ?

File: apis/v1alpha1/ack-generate-metadata.yaml (hunk 1 above)
Comment (vflaux) [reply apis/v1alpha1/ack-generate-metadata.yaml:5]: I didn't made any changes to the code-generator. But I built it from the main branch.

**Comment by a-hilaly** [comment]
@vflaux the current "issue" with the sdk.go should be fixed in https://github.com/aws-controllers-k8s/code-generator/pull/489 - please pull the latest code-gen changes and regenerate the controller. Happy to lgtm right away

**Comment by a-hilaly** [comment]
Also, the PR is still in draft state

**Pull request merged**